typed-racket-lib/typed-racket/typecheck/tc-toplevel.rkt

```diff
@@ -400,6 +400,8 @@
           [else (int-err "Two conflicting definitions: ~a ~a" def other-def)]))
       (dict-update h (binding-name def) merge-def-bindings #f)))
   (do-time "computed def-tbl")
+  ;; check that all parsed apps are sensible:
+  (check-registered-apps!)
```

Comment:
There are plenty of other places where types get parsed. Are we sure that this is after all of them in this code? And the check probably needs to be added elsewhere.